plugin.video.srgssr_ch_replay/resources/lib/plugin.py

```diff
@@ -0,0 +1,536 @@
+        resource_list = media_composition["chapterList"][0]["resourceList"]
+        sd_hls_resources = []
+        for resource in resource_list:
+            if resource["protocol"] == "HLS":
+                if resource["quality"] == "HD":
+                    return resource
+                else:
+                    sd_hls_resources.append(resource)
+
+        if not sd_hls_resources:
+            return resource_list[0]
+        else:
+            return sd_hls_resources[0]
+
+    def _get_media_url(self, resource_url: str) -> str:
+        """Parses the resource URL and constructs the media URL from it"""
+        parsed_url = urlparse(resource_url)
+        media_url = f"{parsed_url.scheme}://{parsed_url.netloc}{parsed_url.path}"
+
+        # add authentication token for akamaihd
+        if "akamaihd" in parsed_url.netloc:
+            self.logger.debug("AkamaiHD video")
+            token_url = f"http://tp.srgssr.ch/akahd/token?acl={parsed_url.path}"
+            response = requests.get(token_url).json()
+            token = response["token"]["authparams"]
+            media_url += "?" + token
+        return media_url
+
+    def _set_inputstream_params(self, listitem, protocol, mime_type):
+        """If Inputstream Adaptive is enabled and available, configure it and update the ListItem"""
+        is_helper = inputstreamhelper.Helper(protocol)
+        if to_bool(self.settings.enable_inputstream_adaptive) and is_helper.check_inputstream():
+            listitem.setContentLookup(False)
+            listitem.setMimeType(mime_type)
+            listitem.setProperty("inputstream", is_helper.inputstream_addon)
+            listitem.setProperty("inputstream.adaptive.manifest_type", protocol)
+
+    def _add_subtitles(self, listitem, video_id):
+        """If subtitles are enable and available, add them to the ListItem"""
+        self.logger.debug(f"Getting subtitles for video {video_id}")
+        if to_bool(self.settings.enable_subtitles):
+            video_urn = f"urn:{self.bu}:episode:tv:{video_id}"
+            resp = self.subs_client.get_subtitles(video_urn)
+
+            subs = []
+            for asset in resp["data"]["assets"]:
+                if asset is not None:
+                    for sub in asset["hasSubtitling"]:
+                        subs.append(sub["identifier"])
+            if subs:
+                self.logger.debug(f"Found subtitles: {subs}")
+                listitem.setSubtitles(subs)
+
+    # ================================= Helper methods ==================================
+
+    def _add_video_to_directory(self, show: dict, episode: dict, media: dict):
+        """Helper that adds a "Video" item to the Directory"""
+        url_args = {
+            "video_id": episode.get("id"),
+            "media_id": media.get("id"),
+        }
+
+        vid_name = episode.get("title", "") + " - " + media.get("title", "") if episode.get("title", "") != media.get("title", "") else episode.get("title", "")
```

Comment:
I changed this line as well to reduce visual redundancy. 
I'm done so far from my side. If you are happy as well with the current state, then it can be merged. Thank you for your help.